cli/internal/helm/overrides.go

```diff
@@ -0,0 +1,156 @@
+// Also, the charts are not rendered correctly without all of these values.
+func extraCiliumValues(provider cloudprovider.Provider, conformanceMode bool, output terraform.ApplyOutput) map[string]any {
+	extraVals := map[string]any{}
+	if conformanceMode {
+		extraVals["kubeProxyReplacementHealthzBindAddr"] = ""
+		extraVals["kubeProxyReplacement"] = "partial"
+		extraVals["sessionAffinity"] = true
+		extraVals["cni"] = map[string]any{
+			"chainingMode": "portmap",
+		}
+	}
+
+	extraVals["k8sServiceHost"] = output.IP
+	extraVals["k8sServicePort"] = 6443 // TODO take from tf?
+	if provider == cloudprovider.GCP {
+		extraVals["ipv4NativeRoutingCIDR"] = output.GCP.IPCidrPod
+		extraVals["strictModeCIDR"] = output.GCP.IPCidrPod
+	}
+	return extraVals
+}
+
+// extraConstellationServicesValues extends the given values map by some values depending on user input.
+// Values set inside this function are only applied during init, not during upgrade.
+func extraConstellationServicesValues(cfg *config.Config, masterSecret, salt []byte, uid, serviceAccURI string, output terraform.ApplyOutput,
+) (map[string]any, error) {
+	attestationConfigJSON, err := json.Marshal(cfg.GetAttestationConfig())
+	if err != nil {
+		return nil, fmt.Errorf("marshalling measurements: %w", err)
+	}
+	extraVals := map[string]any{}
+	extraVals["join-service"] = map[string]any{
+		"measurementSalt":    base64.StdEncoding.EncodeToString(salt),
+		"attestationVariant": cfg.GetAttestationConfig().GetVariant().String(),
+		"attestationConfig":  string(attestationConfigJSON),
+	}
+	extraVals["verification-service"] = map[string]any{
+		"attestationVariant": cfg.GetAttestationConfig().GetVariant().String(),
+		"loadBalancerIP":     output.IP,
+	}
+	extraVals["konnectivity"] = map[string]any{
+		"loadBalancerIP": output.IP,
+	}
+
+	extraVals["key-service"] = map[string]any{
+		"masterSecret": base64.StdEncoding.EncodeToString(masterSecret),
+		"salt":         base64.StdEncoding.EncodeToString(salt),
+	}
+	switch cfg.GetProvider() {
+	case cloudprovider.OpenStack:
+		extraVals["openstack"] = map[string]any{
+			"deployYawolLoadBalancer": cfg.DeployYawolLoadBalancer(),
+		}
+		if cfg.DeployYawolLoadBalancer() {
+			extraVals["yawol-controller"] = map[string]any{
+				"yawolOSSecretName": "yawolkey",
+				// has to be larger than ~30s to account for slow OpenStack API calls.
+				"openstackTimeout": "1m",
+				"yawolFloatingID":  cfg.Provider.OpenStack.FloatingIPPoolID,
+				"yawolFlavorID":    cfg.Provider.OpenStack.YawolFlavorID,
+				"yawolImageID":     cfg.Provider.OpenStack.YawolImageID,
+			}
+		}
+	case cloudprovider.GCP:
+		serviceAccountKey, err := gcpshared.ServiceAccountKeyFromURI(serviceAccURI)
+		if err != nil {
+			return nil, fmt.Errorf("getting service account key: %w", err)
+		}
+		rawKey, err := json.Marshal(serviceAccountKey)
+		if err != nil {
+			return nil, fmt.Errorf("marshaling service account key: %w", err)
+		}
```

Comment:
Could you move this marshalling up into `init.go`? This would remove the `gcpshared` import.